frame/staking/src/lib.rs

```diff
@@ -2235,9 +2092,16 @@ decl_module! {
 		///    minimized (to ensure less variance)
 		///
 		/// # <weight>
-		/// See `crate::weight` module.
+		/// The transaction is assumed to be the longest path, a better solution.
+		///   - Initial solution is almost the same.
+		///   - Worse solution is retraced in pre-dispatch-checks which sets its own weight.
 		/// # </weight>
-		#[weight = weight::weight_for_submit_solution::<T>(winners, compact, size)]
+		#[weight = T::WeightInfo::submit_solution_better(
+			size.validators as u32,
+			size.nominators as u32,
```

Comment:
```suggestion
			size.validators.into(),
			size.nominators.into(),
```

Reply:
`u32` is not `From<usize>` AFAIK, but we make this assumption almost always that usize == u32 in wasm.

Reply:
if I read correctly nominators is u32 and validators is u16.

This change is not important, just if somehow NominatorsIndex is switched to u64 then user can do a wrap around in the conversion.

Reply:
ah right I misread 🤦